### PR TITLE
Introducing Localization with Docusaurus and Crowdin

### PR DESCRIPTION
**Pull request by ImmaZoni**
# 🌍 Introduction to Localization
With the goal of expanding our global reach, we are introducing localization to our project. Localization is the process of adapting content to specific regions and cultures. It's an essential step in making our documentation more accessible to a broader audience.

# 📘 Docusaurus Localization
For the localization process, we are using Docusaurus, a modern static website generator. Docusaurus provides built-in support for translating and versioning our documentation. This PR will close issue #22 

**Here's an overview of the changes and improvements:**

- Language Configuration: Added 16 different locales to the docusaurus.config.js file to support multiple languages.
- Markdown Translation: Enabled translation for all Markdown files within the /docs and /versioned_docs directories.
- Automated Script: Created a script to copy documentation content to the corresponding language directories, preserving the structure for both current and versioned documents.

## 🌐 Crowdin Integration
Crowdin is a cloud-based solution that streamlines localization management. We've integrated Crowdin into our project to facilitate the translation process.

**Here's how Crowdin works with our setup:**
- **API Integration**: Used Crowdin API tokens to automate the uploading and downloading of translation files.
- **YAML Configuration:** Created a Crowdin configuration file to map the source files and the translated content.
- **Environment Variables:** Utilized environment variables to securely manage API keys, ensuring a safe production and development environment.

_Note: Crowdin' Automated CI has been giving us some issues, as such we have a script called reorganize which has helped automate this process. This is a temporary workaround that has been added to CI until we are able to resolve the happy path with Crowdin._

## 📝 How to Contribute to Translation
All team members can contribute to translations through the Crowdin platform. Simply log in, choose the language you want to translate, and start contributing. Your contributions will be synchronized with our codebase.

You can view our translation tutorial here: https://github.com/subspace/subspace-docs/blob/main/TRANSLATE.md

## 🚀 Impact
These localization improvements are a significant step toward engaging with a global audience. They will make our content more accessible, inclusive, and user-friendly.

## 🧪 Testing
Thorough testing has been conducted to ensure that the translation mechanism works seamlessly across different languages and versions.

📚 Further Reading
[Docusaurus Localization Guide](https://docusaurus.io/docs/i18n/introduction)
[Crowdin Official Documentation](https://support.crowdin.com/)

### CODE REVIEW

**Review by jim-counter — APPROVED** [review]
LGTM Justin. Language navigation works and I can see some French localization is already set on some pages. We may find problems as we get contributions but mechanically I couldn't find any showstoppers. Great work!

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -63,8 +71,9 @@ b) Run the dev server for each language with `yarn start -- --locale en` (Replac
 
 ### Deploy (Staff Use Only)
 
-This is only for preparing the documentation for deployment to GitHub Pages.
+While the section below may still be useful, its worth noting that we have automated this process with github actions, now a build we automatically be deployed anytime a push is made to main, aswell as once every 24hr's
```
Comment (EmilFattakhov) [inline DEVELOPMENT.md:74]: its -> it's
we automatically -> will automatically
aswell -> as well
not sure about 24hr's, maybe 24hrs?

**Review by EmilFattakhov — APPROVED** [review]
Awesome work Justin, looks great!

Add a few rewording suggestions, but localization works very well!

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -2,45 +2,53 @@
...
+
+Overall, this script facilitates the handling of the latest version in the Docusaurus documentation system, streamlining the process of deletion, creation, and modification as required. It ensures that the documentation maintains a consistent and controlled structure across different versions.
+
+### Localization
+
+Localization is managed with the [i18n docusaurus plugin](https://docusaurus.io/docs/versioning) in conjunction with the [Crowdin translation portal](https://support.crowdin.com/translation-process-overview/).
+
+**Our localization works in two parts**
+
+1. **i18n Plugin**: This plugin loads translated data for each language and allows the user to select which language they prefer. This plugin also handles all of the routing, and [code adjustment for static react pages](https://docusaurus.io/docs/i18n/tutorial#translate-your-react-code) on our documentation, aswell as [generate .json string](https://docusaurus.io/docs/i18n/tutorial#translate-plugin-data) files for other non-markdown pages. Additoinally it will help us in managing [translated .md files](https://docusaurus.io/docs/i18n/tutorial#translate-markdown-files)
```
[the remark this answers is not in the document]
Comment (ImmaZoni) [reply DEVELOPMENT.md:49]: fixed the aswell, the additionally parts are both start of sentences so they will remain capitalized

File: DEVELOPMENT.md (hunk 1 above)
Comment (ImmaZoni) [reply DEVELOPMENT.md:74]: Fixed in latest commit 😄

**Pull request merged**